### PR TITLE
docs(wildfly): document wildfly 27 and jakarta support

### PR DESCRIPTION
**Pull request by tasso94**
related to camunda/camunda-bpm-platform#3101

### CODE REVIEW

File: content/installation/full/jboss/manual.md [hunk 1]
```diff
@@ -44,15 +47,15 @@ $WILDFLY_HOME/modules/system/layers/base/com/h2database
...
+### Adjust the Configuration
 
 Next, a number of changes need to be performed in the application server's configuration file.
 In most cases this is `$WILDFLY_HOME/standalone/configuration/standalone.xml`.
 
 Add the Camunda subsystem as extension:
 
 ```xml
-<server xmlns="urn:jboss:domain:2.1">
+<server xmlns="urn:jboss:domain:20.0">
```
Comment (tmetzke) [inline content/installation/full/jboss/manual.md:58]: 👍 Nice catch!

File: content/installation/full/jboss/manual.md [hunk 2]
```diff
@@ -144,33 +147,34 @@ These links point you to resources for other databases:
 * [How to configure an Oracle database](http://www.ironjacamar.org/doc/userguide/1.0/en-US/html_single/#ex_datasources_oracle)
 * [How to configure a MySQL database](http://www.ironjacamar.org/doc/userguide/1.0/en-US/html_single/#ex_datasources_mysql)
 
+## Optional Components
 
-# Optional Components
-
-This section describes how to install optional dependencies. None of these are required to work with the core platform. Before continuing, make sure that the Camunda Platform is already installed according to [this step]({{< relref "#required-setup-for-wildfly-jboss-eap-7" >}}) for WildFly / JBoss EAP 7.
+This section describes how to install optional dependencies. None of these are required to work with the core platform. Before continuing, make sure that the Camunda Platform is already installed according to [this step]({{< relref "#required-setup" >}}) for WildFly / JBoss EAP 7.
```
Comment (tmetzke) [inline content/installation/full/jboss/manual.md:152]: 🔧 If we adjust the required setup section name, we need to adjust it here as well.
```suggestion
This section describes how to install optional dependencies. None of these are required to work with the core platform. Before continuing, make sure that the Camunda Platform is already installed according to [this step]({{< relref "#setup" >}}) for WildFly / JBoss EAP 7.
```

File: content/installation/full/jboss/manual.md [hunk 3]
```diff
@@ -209,17 +213,20 @@ To activate Camunda Connect functionality for a process engine, a process engine
 ```
 
 
-## Camunda Spin
+### Camunda Spin
 
-The Camunda Spin plugin can be use to extend the engine functionality to de-/serialize object variables from and to JSON and XML. For more information, see the [Spin Reference]({{< ref "/reference/spin/_index.md" >}}).
+The Camunda Spin plugin can be used to extend the engine functionality to de-/serialize object variables from and to JSON and XML. For more information, see the [Spin Reference]({{< ref "/reference/spin/_index.md" >}}).
```
Comment (tmetzke) [inline content/installation/full/jboss/manual.md:218]: 🔧 Let's try to use active voice here.
```suggestion
You can use the Camunda Spin plugin to extend the engine functionality to de-/serialize object variables from and to JSON and XML. For more information, see the [Spin Reference]({{< ref "/reference/spin/_index.md" >}}).
```

File: content/update/minor/718-to-719/_index.md [hunk 4]
```diff
@@ -133,3 +134,41 @@ Tenant information is populated for User operation logs from 7.19 onwards, user
 * Adding/Clearing a user operation log annotation
 
 In case you want to avoid tenant check, please refer to [Disable the transparent access restrictions]({{< ref "/user-guide/process-engine/multi-tenancy.md#disable-the-transparent-access-restrictions" >}}).
+
+# Update to Wildfly 27 Application Server
+
+With this release, you can run Camunda 7 as shared-engine on the [WildFly 27 application server](https://www.wildfly.org/news/2022/11/09/WildFly27-Final-Released/).
+Since Wildfly 27 drops support for Java EE in favor of Jakarta EE, you need to migrate your process applications and replace some artifacts on the application server.
```
Comment (tmetzke) [inline content/update/minor/718-to-719/_index.md:141]: ❓ Should we hint people at the WildFly 26 modules that we still provide if they don't want to move to WildFly 27 but still update to 7.19.x? It currently reads a bit like "Here is 7.19.x, please move to WildFly 27". Maybe we can avoid some questions popping up in Support if we explicitly make clear that moving to WildFly 27 is optional 😃 I was thinking in a direction of "We support WildFly 27 now, it is the new default pre-packaged distro, it moves to Jakarta EE, you can also move or stay on WildFly 26, if you want to move here are the things to consider".

File: content/update/minor/718-to-719/_index.md [hunk 5]
```diff
@@ -133,3 +134,41 @@ Tenant information is populated for User operation logs from 7.19 onwards, user
...
+
+## Migrate process applications
+
+* Replace Java EE class references (`javax.*`) with Jakarta class references (`jakarta.*`)
+  * You might have a look at [`org.eclipse.transformer:transformer-maven-plugin`](https://github.com/eclipse/transformer)
+* Replace Camunda class references:
+  * `org.camunda.bpm.application.impl.EjbProcessApplication` → `org.camunda.bpm.application.impl.JakartaEjbProcessApplication`
+  * `org.camunda.bpm.application.impl.ServletProcessApplicationDeployer` → `org.camunda.bpm.application.impl.JakartaServletProcessApplicationDeployer`
+  * `org.camunda.bpm.application.impl.ServletProcessApplication` → `org.camunda.bpm.application.impl.JakartaServletProcessApplication`
+  * `org.camunda.bpm.engine.impl.cfg.jta.JtaTransactionContextFactory` → `org.camunda.bpm.engine.impl.cfg.jta.JakartaTransactionContext`
```
Comment (tmetzke) [inline content/update/minor/718-to-719/_index.md:151]: ❌ Class mixup
```suggestion
  * `org.camunda.bpm.engine.impl.cfg.jta.JtaTransactionContextFactory` → `org.camunda.bpm.engine.impl.cfg.jta.JakartaTransactionContextFactory`
```

File: content/update/minor/718-to-719/_index.md [hunk 6]
```diff
@@ -133,3 +134,41 @@ Tenant information is populated for User operation logs from 7.19 onwards, user
...
+  * `org.camunda.bpm.engine.impl.cfg.jta.JtaTransactionContext` → `org.camunda.bpm.engine.impl.cfg.jta.JakartaTransactionContextFactory`
```
Comment (tmetzke) [inline content/update/minor/718-to-719/_index.md:152]: ❌ Class mixup
```suggestion
  * `org.camunda.bpm.engine.impl.cfg.jta.JtaTransactionContext` → `org.camunda.bpm.engine.impl.cfg.jta.JakartaTransactionContext`
```

File: content/update/minor/718-to-719/jboss.md [hunk 7]
```diff
@@ -14,9 +14,8 @@ The following steps describe how to update the Camunda artifacts on a Wildfly/JB
 shared process engine scenario. Throughout the procedure, refer to the [update guide][update-guide].
 
 If not already done, download the [Camunda Platform 7.19 JBoss distribution](https://downloads.camunda.cloud/release/camunda-bpm/jboss/7.19/)
```
Comment (tmetzke) [inline content/update/minor/718-to-719/jboss.md:16]: ❌ This should be WildFly distribution, I think.
```suggestion
If not already done, download the [Camunda Platform 7.19 WildFly distribution](https://downloads.camunda.cloud/release/camunda-bpm/wildfly/7.19/)
```

File: content/update/minor/718-to-719/jboss.md [hunk 8]
```diff
@@ -14,9 +14,8 @@ The following steps describe how to update the Camunda artifacts on a Wildfly/JB
...
-or [Camunda Platform 7.19 Wildfly distribution](https://downloads.camunda.cloud/release/camunda-bpm/wildfly/7.19/). In the following instructions,
-`$APP_SERVER` should be replaced with either `jboss` or `wildfly`, depending on
-the used application server.
+or [Camunda Platform 7.19 Wildfly ≤26 Modules](https://artifacts.camunda.com/artifactory/camunda-bpm/org/camunda/bpm/wildfly/camunda-wildfly26-modules/). 
+In the following instructions, `$APP_SERVER` should be replaced with either `jboss` or `wildfly`, depending on the used application server.
```
Comment (tmetzke) [inline content/update/minor/718-to-719/jboss.md:18]: ❓ Does the `$APP_SERVER` variable replacement still make sense? We don't offer a JBoss subsystem anymore, do we?

File: content/user-guide/process-applications/the-process-application-class.md [hunk 9]
```diff
@@ -330,14 +367,14 @@ The following shows an example of how to bootstrap a SpringProcessApplication in
 </beans>
 ```
 
-(Remember that you additionally need a `META-INF/processes.xml` file.
+Remember that you additionally need a `META-INF/processes.xml` file.
```
Comment (tmetzke) [inline content/user-guide/process-applications/the-process-application-class.md:370]: 🔧 We can rewrite this a bit, I think.
```suggestion
Remember to additionally add a `META-INF/processes.xml` file.
```

**Review by tmetzke — CHANGES_REQUESTED** [review]
👍 Love the adjustments.
I add a couple of 🔧 suggestions and ❓ questions where I saw fit. Nothing major in my regard, the overall changes look really good already! 🙂

File: content/installation/full/jboss/manual.md [hunk 10]
```diff
@@ -13,22 +13,25 @@ menu:
...
-This document describes the installation of Camunda Platform and its components on a vanilla [Wildfly Application Server / JBoss EAP 7](http://www.wildfly.org).
+This document describes the installation of Camunda Platform and its components on a vanilla [Wildfly Application Server](http://www.wildfly.org) or JBoss EAP 7.
 
 {{< note title="Reading this Guide" class="info" >}}
 This guide uses a number of variables to denote common path names and constants:
 `$WILDFLY_HOME` points to the JBoss EAP/Wildfly application server main directory.
 `$PLATFORM_VERSION` denotes the version of the Camunda Platform you want to install or already have installed, e.g. `7.0.0`.
 {{< /note >}}
 
+## Required Setup
```
[the remark this answers is not in the document]
Comment (tasso94) [reply content/installation/full/jboss/manual.md:24]: 👍

File: content/installation/full/jboss/manual.md [hunk 11]
```diff
@@ -13,22 +13,25 @@ menu:
...
+### Wildfly 27+
 
-This section explains how to perform the required setup steps for Wildfly Application Server.
+For Wildfly 27  and above, please download the [Camunda WildFly distribution](https://downloads.camunda.cloud/release/camunda-bpm/wildfly/).
 
-First, you need to download the [Camunda WildFly distribution](https://downloads.camunda.cloud/release/camunda-bpm/wildfly/).
+### Wildfly ≤26 / JBoss EAP 7
 
-## Copy Modules
+For Wildfly ≤26 / JBoss EAP 7, please download the [`camunda-wildfly26-modules`](https://artifacts.camunda.com/artifactory/camunda-bpm/org/camunda/bpm/wildfly/camunda-wildfly26-modules/).
```
[the remark this answers is not in the document]
Comment (tasso94) [reply content/installation/full/jboss/manual.md:32]: 👍

File: content/installation/full/jboss/manual.md [hunk 12]
```diff
@@ -13,22 +13,25 @@ menu:
 ---
 
 
-This document describes the installation of Camunda Platform and its components on a vanilla [Wildfly Application Server / JBoss EAP 7](http://www.wildfly.org).
+This document describes the installation of Camunda Platform and its components on a vanilla [Wildfly Application Server](http://www.wildfly.org) or JBoss EAP 7.
 
 {{< note title="Reading this Guide" class="info" >}}
 This guide uses a number of variables to denote common path names and constants:
 `$WILDFLY_HOME` points to the JBoss EAP/Wildfly application server main directory.
 `$PLATFORM_VERSION` denotes the version of the Camunda Platform you want to install or already have installed, e.g. `7.0.0`.
```
[the remark this answers is not in the document]
Comment (tasso94) [reply content/installation/full/jboss/manual.md:21]: 👍

File: content/installation/full/jboss/manual.md (hunk 3 above)
Comment (tasso94) [reply content/installation/full/jboss/manual.md:218]: 👍

File: content/update/minor/718-to-719/_index.md (hunk 5 above)
Comment (tasso94) [reply content/update/minor/718-to-719/_index.md:151]: 👍

File: content/update/minor/718-to-719/_index.md (hunk 6 above)
Comment (tasso94) [reply content/update/minor/718-to-719/_index.md:152]: 👍

File: content/user-guide/process-applications/the-process-application-class.md (hunk 9 above)
Comment (tasso94) [reply content/user-guide/process-applications/the-process-application-class.md:370]: 👍

File: content/update/minor/718-to-719/_index.md (hunk 4 above)
Comment (tasso94) [reply content/update/minor/718-to-719/_index.md:141]: 👍

**Comment by tasso94** [comment]
Thanks for the thorough review! Great hints 👍

**Review by tmetzke — APPROVED** [review]
👍 Marvelous, this looks really good 🙂 
Shall we involve a tech writer review as well?

**Review by christinaausley — APPROVED** [review]
Technical review in latest commit, largely adjusting Wildfly to WildFly 👍

**Comment by tasso94** [comment]
Great team work! Thank you, @christinaausley and @tmetzke 🙂 👍

**Pull request merged**